src/NamedTupleUtilities.jl

```diff
@@ -91,48 +122,107 @@ ERROR: duplicate field name in NamedTuple: "c" is not unique
+Return a NamedTuple which retains the fields with names ending with `b` in `a`. 
 ```jldoctest
-julia> endswith((abc=1,bcd=2,cde=3),Val(:d))
+julia> QueryOperators.NamedTupleUtilities.endswith((abc=1,bcd=2,cde=3),Val(:d))
 (bcd = 2,)
 ```
 """
 @generated function endswith(a::NamedTuple{an}, ::Val{bn}) where {an, bn}
     names = ((i for i in an if Base.endswith(String(i), String(bn)))...,)
-    types = Tuple{(fieldtype(a ,n) for n in names)...}
+    types = Tuple{(fieldtype(a, n) for n in names)...}
+    vals = Expr[:(getfield(a, $(QuoteNode(n)))) for n in names]
+    return :(NamedTuple{$names,$types}(($(vals...),)))
+end
+
+"""
+    endswith(a::NamedTuple, b::Val{n})
+Return a NamedTuple which retains the fields with names that do not end with `b` in `a`. 
+```jldoctest
+julia> QueryOperators.NamedTupleUtilities.not_endswith((abc=1,bcd=2,cde=3),Val(:d))
+(abc = 1, cde = 3)
+```
+"""
+@generated function not_endswith(a::NamedTuple{an}, ::Val{bn}) where {an, bn}
+    names = ((i for i in an if !Base.endswith(String(i), String(bn)))...,)
+    types = Tuple{(fieldtype(a, n) for n in names)...}
     vals = Expr[:(getfield(a, $(QuoteNode(n)))) for n in names]
     return :(NamedTuple{$names,$types}(($(vals...),)))
 end
 
 """
     occursin(a::NamedTuple, b::Val{n})
-Return a NamedTuple which retains the fields with names contains `b` as a substring. 
+Return a NamedTuple which retains the fields with names containing `b` as a substring. 
 ```jldoctest
-julia> occursin((abc=1,bcd=2,cde=3),Val(:d))
+julia> QueryOperators.NamedTupleUtilities.occursin((abc=1,bcd=2,cde=3),Val(:d))
 (bcd = 2, cde = 3)
 ```
 """
 @generated function occursin(a::NamedTuple{an}, ::Val{bn}) where {an, bn}
     names = ((i for i in an if Base.occursin(String(bn), String(i)))...,)
-    types = Tuple{(fieldtype(a ,n) for n in names)...}
+    types = Tuple{(fieldtype(a, n) for n in names)...}
+    vals = Expr[:(getfield(a, $(QuoteNode(n)))) for n in names]
+    return :(NamedTuple{$names,$types}(($(vals...),)))
+end
+
+"""
+    not_occursin(a::NamedTuple, b::Val{n})
+Return a NamedTuple which retains the fields without names containing `b` as a substring. 
+```jldoctest
+julia> QueryOperators.NamedTupleUtilities.not_occursin((abc=1,bcd=2,cde=3),Val(:d))
+(abc = 1,)
+```
+"""
+@generated function not_occursin(a::NamedTuple{an}, ::Val{bn}) where {an, bn}
+    names = ((i for i in an if !Base.occursin(String(bn), String(i)))...,)
+    types = Tuple{(fieldtype(a, n) for n in names)...}
+    vals = Expr[:(getfield(a, $(QuoteNode(n)))) for n in names]
+    return :(NamedTuple{$names,$types}(($(vals...),)))
+end
+
+"""
+    oftype(a::NamedTuple, b::DataType)
+Returns a NamedTuple which retains the fields whose elements have type `b`.
+```jldoctest
+julia> QueryOperators.NamedTupleUtilities.oftype((a = [4,5,6], b = [3.,2.,1.], c = ["He","llo","World!"]), Int64)
+(a = [4, 5, 6],)
+```
+"""
+@generated function oftype(a::NamedTuple{an}, b::DataType) where {an}
```

Comment:
Does this actually work? I would have thought we need to do something like ``::Val{b}`` here?